### ISSUE TITLE
Referral Rewards

### ISSUE THREAD

**Issue by calaber24p** [dev-protocol/DIPs#21]
Referral rewards could greatly increase the amount of creators moving to the platform by offering an incentive to dev users. Under this proposal 1%-5% of the rewards from onboarded creators through the referral system, will be rewarded to the users who bring them to the platform. 

Users can use their Eth addresses as a form of identification when referring new creators to the platform. Once a project that the user refers is deemed to be legitimate and is brought online, they will earn a percentage of their earnings for a fixed period of time. Initially a believe a three month period would be a good number. 

The amount they are rewarded scales with the amount staked on the property, so bringing bigger creators to the platform will offer larger rewards for those looking to refer them. One of the larger problems we are facing right now is bringing new users to the platform and I believe that this system could make this process much easier and save us a large amount of time. 

Ultimately we still have the final approval on whether or not the project is legitimate and should be approved through the manual review system. The only downside I can see is the possibility of referral disputes from users who might bring a creator to the platform but that creator doesn't state the user brought them there.

**Comment by aggre** [comment]
I think it's an important proposal.

I think it would be better if there was an incentive for creators to accept referrals by referrers.

My idea for that is for referrers to deposit their own DEV in advance into a contract for the referral program (Turnstile contract), and then automatically staking the deposited DEV when creators entered.

In technically we can build these programs using the following elements:

- "Turnstile contract": a contract that allows creators to automatically deposit a portion (1-5%) of a newly created Property (ERC20) and bypass it to the referrer. After three months, creators can get back the deposited Property share in this contract to itself.
- "Dapp for referral program": a web page with a friendly interface for creators to participate in the program.

**Comment by aggre** [comment]
@Akira-Taniguchi Can you consider the specs and interfaces for the Turnstile contract?

My concerns:
- The `PropertyFactory.createAndAuthenticate` function treats the transaction sender as the Property author. Creators must pre-call the `PropertyFactory.createAndAuthenticate` function, as it is not desirable for the Turnstile contract to be the Property author.
- At the time of successful authentication (when `Market.authenticatedCallback` is executed), additional processing cannot be hooked. Therefore, the following process requires a transaction from the Property author.
  - Send a share of the Property (1-5%) to the Turnstile contract. (The Turnstile Contract holds the share, but the referrer receives the creator's reward.)
  - Staking on the Property, the DEV deposited by the referrer to the Turnstile contract. (using `Dev.depositFrom` function).

**Comment by Akira-Taniguchi** [comment]
>Send a share of the Property (1-5%) to the Turnstile contract. (The Turnstile Contract holds the share, but the referrer receives the creator's reward.)
>Staking on the Property the DEV deposited by the referrer to the Turnstile contract. (using Dev.depositFrom function).

I think this specification will work.
I think the concern is what variable in AddressConfig will manage the address of the turnstile contract.

@aggre

**Comment by aggre** [comment]
@Akira-Taniguchi Great. Can you design it as an external contract?

**Comment by aggre** [comment]
I noticed that if the referrer stake the target Property, that would be self-staking. We will look into whether we can design incentives for creators in a different scheme.

**Comment by Akira-Taniguchi** [comment]
It's probably simpler to simply give the referrer a DEV token instead of getting property or automatic staking or something like that.

**Comment by defi-er** [comment]
@Akira-Taniguchi I agree. I think that the incubator will be a better growth model than a referral program.

**Comment by calaber24p** [comment]
Yeah the referrer would not get a percentage of the property, but they would get the amount that property would earn in dev. Whatever is easiest to calculate that way would be best. @defi-er most likely the incubator wont give rewards to users who want to onboard people, only to the company being incubated. The referral system would allow regular users to earn extra dev by putting in the work

**Comment by defi-er** [comment]
A better use of time is building the incubator structure which gives a stronger value proposition to projects rather than turning users into sales men with no sales collateral.

**Comment by calaber24p** [comment]
It was an idea some people had asked for so I proposed it. I agree the incubator is probably a better use of the current time, but maybe in the future a referral system can be added if there' less of a workload.

**Comment by aggre** [comment]
So let's prioritise the Dev Incubation Program.
